src/robot/ros/RosRobot.cpp

```diff
@@ -38,6 +52,262 @@ RosRobot::RosRobot(
+  deactivateExecutor();
+
+  // Validate input
+  if (mExecutors.find(id) == mExecutors.end())
+  {
+    return false;
+  }
+
+  if (!mRosControllerNames[id].empty()
+      && !startController(mRosControllerNames[id]))
+  {
+    throw std::runtime_error(
+        "Could not start controller: " + mRosControllerNames[id]);
+  }
+
+  if (mRosControllerModes[id]
+          != hardware_interface::JointCommandModes::ERROR
+      && !switchControllerMode(mRosControllerModes[id]))
+  {
+    throw std::runtime_error(
+        "Could not switch controller mode to: "
+        + aikido::control::ros::intFromMode(
+            mRosControllerModes[id]));
+  }
+
+  return Robot::activateExecutor(id);
+}
+
+//=============================================================================
+void RosRobot::setRosLoadControllerServiceClient(
+      const std::shared_ptr<::ros::ServiceClient>& rosLoadControllerServiceClient)
+{
+  mRosLoadControllerServiceClient = rosLoadControllerServiceClient;
+}
+
+//=============================================================================
+void RosRobot::setRosSwitchControllerServiceClient(
+      const std::shared_ptr<::ros::ServiceClient>& rosSwitchControllerServiceClient)
+{
+  mRosSwitchControllerServiceClient = rosSwitchControllerServiceClient;
+}
+
+//=============================================================================
+void RosRobot::setRosJointModeCommandClient(
+      const std::shared_ptr<aikido::control::ros::RosJointModeCommandClient>& rosJointModeCommandClient)
+{
+  mRosJointModeCommandClient = rosJointModeCommandClient;
+}
+
+//=============================================================================
+bool RosRobot::loadController(const std::string loadControllerName)
+{
+  if (!mRosLoadControllerServiceClient)
+  {
+    throw std::runtime_error(
+        "ROS controller manager service client not instantiated.");
+  }
+
+  controller_manager_msgs::LoadController srv;
+  srv.request.name = loadControllerName;
+
+  return mRosLoadControllerServiceClient->call(srv) && srv.response.ok;
+}
+
+//=============================================================================
+bool RosRobot::startController(const std::string startControllerName)
+{
+  return switchControllers(std::vector<std::string>{startControllerName}, std::vector<std::string>{});
+}
+
+//=============================================================================
+bool RosRobot::stopController(const std::string stopControllerName)
+{
+  cancelAllCommands();
+  return switchControllers(std::vector<std::string>{}, std::vector<std::string>{stopControllerName});
+}
+
+//=============================================================================
+bool RosRobot::switchControllers(
+    const std::vector<std::string> startControllersNames, const std::vector<std::string> stopControllersNames)
+{
+  if (!mRosSwitchControllerServiceClient)
+  {
+    throw std::runtime_error(
+        "ROS controller manager service client not instantiated.");
```

Comment:
Minimize the use of runtime errors when not necessary. Print a warning and return false.